winbuild/pthread.c

```diff
@@ -69,6 +69,21 @@ static void _pthread_once_cleanup(pthread_once_t *o)
   *o = 0;
 }
 
+/* Ensure the CriticalSection has been initialized */
+static void ensure_mutex_init(pthread_mutex_t *m) {
+   if (m->initialized) return;
+
+   if (!m->initialized) {
```

Comment:
this is racy; if two threads are calling in here at the same time, one may initialize and lock the critical section before the other initializes and then successfully locks it; that was the reason behind spinlocking.

An alternative to fixing up the spinlock is using a constructor function to initialize a global mutex in here just for guarding this bit of code.

I can't comment more right now, but take a look at `w_ctor_fn_reg` for registering a constructor